### PR TITLE
Update E0271 explanation

### PR DESCRIPTION
**Pull request by strom-und-spiele**
When reading the explanation, I got confused by it. I hope the wording is clearer now.

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0271.md [hunk 1]
```diff
@@ -56,11 +51,9 @@ foo(3_i8);
 // therefore the type-checker complains with this error code.
 ```
 
-To avoid those issues, you have to make the types match correctly.
-So we can fix the previous examples like this:
-
+The issue can be resolved by changing the associated type
```
Comment (JohnTitor) [inline src/librustc_error_codes/error_codes/E0271.md:54]:
```suggestion
The issue can be resolved by changing the associated type:
```

File: src/librustc_error_codes/error_codes/E0271.md [hunk 2]
```diff
@@ -56,11 +51,9 @@ foo(3_i8);
...
+1) in the `foo` implementation
```
Comment (JohnTitor) [inline src/librustc_error_codes/error_codes/E0271.md:55]:
```suggestion
1) in the `foo` implementation:
```

File: src/librustc_error_codes/error_codes/E0271.md [hunk 3]
```diff
@@ -70,13 +63,17 @@ fn foo<T>(t: T) where T: Trait<AssociatedType = &'static str> {
 impl Trait for i8 { type AssociatedType = &'static str; }
 
 foo(3_i8);
+```
+
+2) in the `Trait` implementation for `i8`
```
Comment (JohnTitor) [inline src/librustc_error_codes/error_codes/E0271.md:68]:
```suggestion
2) in the `Trait` implementation for `i8`:
```

File: src/librustc_error_codes/error_codes/E0271.md [hunk 4]
```diff
@@ -70,13 +63,17 @@ fn foo<T>(t: T) where T: Trait<AssociatedType = &'static str> {
...
+trait Trait { type AssociatedType; }
 
-// For-Loop Example:
-let vs = vec![1, 2, 3, 4];
-for v in &vs {
-    match v {
-        &1 => {}
-        _ => {}
-    }
+fn foo<T>(_t: T) where T: Trait<AssociatedType = u32> {
```
Comment (JohnTitor) [inline src/librustc_error_codes/error_codes/E0271.md:72]:
```suggestion
fn foo<T>(t: T) where T: Trait<AssociatedType = u32> {
```

File: src/librustc_error_codes/error_codes/E0271.md [hunk 5]
```diff
@@ -14,11 +14,6 @@ impl Trait for i8 { type AssociatedType = &'static str; }
 foo(3_i8);
 ```
 
-This is because of a type mismatch between the associated type of some
-trait (e.g., `T::Bar`, where `T` implements `trait Quux { type Bar; }`)
-and another type `U` that is required to be equal to `T::Bar`, but is not.
-Examples follow.
```
Comment (JohnTitor) [inline src/librustc_error_codes/error_codes/E0271.md:20]: Why was this removed, i.e., how did it make you confused?

**Review by JohnTitor — CHANGES_REQUESTED** [review]

File: src/librustc_error_codes/error_codes/E0271.md (hunk 5 above)
Comment (strom-und-spiele) [reply src/librustc_error_codes/error_codes/E0271.md:20]: It felt like this comment was (and it actually is) a left over of an older version. 
Reasons I don't like it:
* It tries to summarize the issue with completely different dummy words.
* It simply states the error message ("This is because of a type mismatch between the associated type of some
trait…)

So basically the first half of the text is contained in the error message itself, the second is explained (better and clearer imho) in the erroneous code example. No need to have this filler text there.

File: src/librustc_error_codes/error_codes/E0271.md (hunk 4 above)
Comment (strom-und-spiele) [reply src/librustc_error_codes/error_codes/E0271.md:72]: The parameter is never used, so it should be prefixed with an underscore. See https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=25a2cafb2b889bac1dbfe58fea32bedb

Comment (JohnTitor) [reply src/librustc_error_codes/error_codes/E0271.md:72]: We don't mind if it's unused or not here. Actually, we don't add underscores to the above snippets and other error codes' snippets.

File: src/librustc_error_codes/error_codes/E0271.md (hunk 5 above)
Comment (JohnTitor) [reply src/librustc_error_codes/error_codes/E0271.md:20]: Okay, it makes sense. Then we could also remove the first snippet? I think it's just a meaningless duplication now.

Comment (strom-und-spiele) [reply src/librustc_error_codes/error_codes/E0271.md:20]: It made me read the code twice and helped me understand the issue better. That's why I left it there. But you are right. Thanks for reviewing :+1:

**Comment by JohnTitor** [comment]
Thanks! I think it's now good to go, r? @Mark-Simulacrum for another approval.

**Comment by Mark-Simulacrum** [comment]
r=me with commits squashed into one

**Comment by strom-und-spiele** [comment]
Sorry I'm not familiar with that `r` shorthand. 
I assume @Mark-Simulacrum is ok with the changes if squashed into one commit.

**Comment by JohnTitor** [comment]
"r=me" means "I'll approve it" so you're right :)
@bors r=Mark-Simulacrum rollup=always

**Comment by strom-und-spiele** [comment]
I like how you resolve my question on one line and create another on the line right after. :D
